src/windows/wsladiag/wsladiag.cpp

```diff
@@ -64,6 +72,59 @@ int wsladiag_main(std::wstring_view commandLine)
         return 0;
     }
 
+    if (!debugShell.empty())
+    {
+        wslutil::PrintMessage(std::format(L"[diag] debugShell='{}'\n", debugShell), stdout);
+
+        try
+        {
+            wil::com_ptr<IWSLAUserSession> userSession;
+            THROW_IF_FAILED(CoCreateInstance(__uuidof(WSLAUserSession), nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(&userSession)));
+
+            wsl::windows::common::security::ConfigureForCOMImpersonation(userSession.get());
+
+            wil::com_ptr<IWSLASession> session;
+
+            // Open the existing session by name/ID from the --debug-shell argument.
+            THROW_IF_FAILED(userSession->OpenSessionByName(debugShell.c_str(), &session));
+
+            wslutil::PrintMessage(L"[diag] OpenSessionByName succeeded\n", stdout);
+
+            std::string shell = "/bin/bash";
```

Comment:
We should probably default to `/bin/sh` since I suspect that bash won't be in our runtime image